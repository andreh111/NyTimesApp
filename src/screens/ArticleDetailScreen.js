import React, {useState} from 'react';
import {Image, View, StyleSheet} from 'react-native';
import {Text} from '@react-native-material/core';
import {useRoute} from '@react-navigation/native';

const ArticleDetailScreen = () => {
  const [imageError, setImageError] = useState(false);
  const {
    params: {title, abstract, published_date, multimedia, isSearching},
  } = useRoute();

  const handleImageError = () => {
    setImageError(true);
  };

  return (
    <View>
      <Image
        style={styles.img}
        source={
          imageError
            ? require('../assets/nytimes.png')
            : {
                uri: isSearching
                  ? `https://static01.nyt.com/${multimedia[0]?.legacy?.xlarge}`
                  : multimedia?.[0]?.url,
              }
        }
        onError={handleImageError}
      />
      <View style={styles.space}>
        <Text style={styles.title}>{title}</Text>
        <Text>{abstract}</Text>
        <Text style={styles.date}>{published_date}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  date: {
    fontSize: 12,
    marginTop: 10,
  },
  img: {
    width: '100%',
    height: 300,
  },
  space: {
    margin: 10,
  },
});

export default ArticleDetailScreen;
