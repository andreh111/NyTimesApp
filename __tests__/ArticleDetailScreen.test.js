import React from 'react';
import renderer from 'react-test-renderer';
import ArticleDetailScreen from '../src/screens/ArticleDetailScreen';

jest.mock('@react-navigation/native', () => ({
  useRoute: () => ({
    params: {
      title: 'Test Title',
      abstract: 'Test Abstract',
      published_date: '2022-01-01',
      multimedia: [{url: 'https://example.com/image.jpg'}],
    },
  }),
}));

describe('ArticleDetailScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ArticleDetailScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
