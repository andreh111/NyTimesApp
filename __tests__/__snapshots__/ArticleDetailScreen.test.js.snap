// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetailScreen renders correctly 1`] = `
<View>
  <Image
    defaultSource={
      {
        "testUri": "../../../src/assets/nytimes.png",
      }
    }
    source={
      {
        "uri": "https://example.com/image.jpg",
      }
    }
    style={
      {
        "height": 300,
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "margin": 10,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          {
            "color": "#000000",
          },
          {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
          },
        ]
      }
    >
      Test Title
    </Text>
    <Text
      style={
        [
          {
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          {
            "color": "#000000",
          },
          undefined,
        ]
      }
    >
      Test Abstract
    </Text>
    <Text
      style={
        [
          {
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          {
            "color": "#000000",
          },
          {
            "fontSize": 12,
            "marginTop": 10,
          },
        ]
      }
    >
      2022-01-01
    </Text>
  </View>
</View>
`;
